contracts/mainnet/connectors/compound-iii/main.sol

```diff
@@ -0,0 +1,907 @@
+	{
+		(uint256 amt_, uint256 setId_) = _borrowOrWithdraw(
+			BorrowWithdrawParams({
+				market: market,
+				token: getBaseToken(market),
+				from: from,
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+		eventName_ = "LogBorrowFrom(address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, from, to, amt_, getId, setId_);
+	}
+
+	/**
+	 * @dev Repays entire borrow of the base asset.
+	 * @notice Repays an entire borrow of the base asset.
+	 * @param market The address of the market.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function payback(address market, uint256 setId)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		require(market != address(0), "invalid market address");
+
+		address token = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		uint256 amt_ = CometInterface(market).borrowBalanceOf(address(this));
+		approve(tokenContract, market, amt_);
+
+		CometInterface(market).supply(token_, uint256(-1));
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogPayback(address,address,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, amt_, setId);
+	}
+
+	/**
+	 * @dev Repays entire borrow of the base asset on behalf of 'to'.
+	 * @notice Repays an entire borrow of the base asset on behalf of 'to'.
+	 * @param market The address of the market.
+	 * @param to The address on behalf of which the borrow is to be repaid.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function paybackOnBehalf(
+		address market,
+		address to,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		require(market != address(0), "invalid market address");
+
+		address token = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		uint256 amt_ = CometInterface(market).borrowBalanceOf(to);
+		approve(tokenContract, market, amt_);
+
+		CometInterface(market).supplyTo(to, token_, uint256(-1));
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogPaybackOnBehalf(address,address,address,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, to, amt_, setId);
+	}
+
+	/**
+	 * @dev Repays entire borrow of the base asset form 'from' on behalf of 'to'.
+	 * @notice Repays an entire borrow of the base asset on behalf of 'to'.
+	 * @param market The address of the market.
+	 * @param from The address from which the borrow has to be repaid on behalf of 'to'.
+	 * @param to The address on behalf of which the borrow is to be repaid.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function paybackFrom(
+		address market,
+		address from,
+		address to,
+		uint256 setId
+	)
```

Comment:
@Richa-iitr Why are we paying back the full amount? We should take `amt` param also, right?

Reply:
resolved